app/jobs/scheduled/bookmark_reminder_notifications.rb

```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+
+module Jobs
+
+  # Runs periodically to send out bookmark reminders, capped at 300 at a time.
+  # Any leftovers will be caught in the next run, because the reminder_at column
+  # is set to NULL once a reminder has been sent.
+  class BookmarkReminderNotifications < ::Jobs::Scheduled
+    MAX_REMINDER_NOTIFICATIONS_PER_RUN = 300
+    JOB_RUN_NUMBER_KEY = 'jobs_bookmark_reminder_notifications_job_run_num'.freeze
+    AT_DESKTOP_CONSISTENCY_RUN_NUMBER = 6
+
+    every 5.minutes
+
+    def execute(args = nil)
+      return if !SiteSetting.enable_bookmarks_with_reminders?
+
+      bookmarks = Bookmark.pending_reminders
+        .where.not(reminder_type: Bookmark.reminder_types[:at_desktop])
+        .includes(:user).order('reminder_at ASC')
+
+      bookmarks.limit(MAX_REMINDER_NOTIFICATIONS_PER_RUN).each do |bookmark|
+        BookmarkReminderNotificationHandler.send_notification(bookmark)
+      end
+
+      # we only want to ensure the desktop consistency every X runs of this job
+      # (every 30 mins in this case) so we don't bother redis too much, and the
+      # at desktop consistency problem should not really happen unless people
+      # are setting the "at desktop" reminder, going out for milk, and never coming
+      # back
+      current_job_run_number = Discourse.redis.get(JOB_RUN_NUMBER_KEY)&.to_i || 0
```

Comment:
```suggestion
      current_job_run_number = Discourse.redis.get(JOB_RUN_NUMBER_KEY).to_i
```
`nil.to_i` returns 0